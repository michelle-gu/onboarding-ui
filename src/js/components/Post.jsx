import React, { Component } from "react";
import ReactDOM from "react-dom";

class Post extends Component {

  	constructor() {
    	super();
  	}

    getCreatedAt() {
      let options = { month: 'short', day: 'numeric' };
      let createdAt  = new Date(this.props.createdAt);
      return createdAt.toLocaleDateString("en-US", options);
    }

  	render() {
    	return (
    		<div className="user">
    			<img id="profile-img" src={this.props.profileImgUrl}></img>
    			<span className="name">{this.props.name}</span>
    			<span className="name small-text">{this.props.twitterHandle}</span>
    		</div>

        <a>
      		<div className="tweet">
      			<div className="date small-text">{this.getCreatedAt()}</div>
      			<div>{this.props.message}</div>
      		</div>
        </a>
    	);
  	}

}

Post.defaultProps = {
	profileImgUrl: "",
  	name: "name",
  	twitterHandle: "twitterHandle",
};

export default Post;
