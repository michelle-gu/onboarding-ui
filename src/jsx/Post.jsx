import React, { Component } from "react";

const options = { month: 'short', day: 'numeric' };

class Post extends Component {

    constructor(props) {
      super(props);

      this.createdAt = new Date(this.props.createdAt).toLocaleDateString("en-US", options);
    }

    render() {
      return (
        <div className="post">
          <div className="user">
            <img id="profile-img" src={this.props.profileImageUrl}></img>
            <span className="name">{this.props.name}</span>
            <span className="name small-text">{this.props.twitterHandle}</span>
          </div>

          <a className="link" href={this.props.url} target="_blank">
            <div className="tweet">
              <div className="date small-text">{this.createdAt}</div>
              <div>{this.props.message}</div>
            </div>
          </a>
        </div>
      );
    }

}

export default Post;
