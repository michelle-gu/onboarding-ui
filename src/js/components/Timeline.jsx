import React, { Component } from "react";
import ReactDOM from "react-dom";
import Post from "./Post.jsx";

class Timeline extends Component {

  	constructor() {
    	super();
  	}

  	render() {
    	return (
      		<Post></Post>
    	);
  	}

}

export default Timeline;

ReactDOM.render(<Timeline />, document.getElementById("timeline-div"));
