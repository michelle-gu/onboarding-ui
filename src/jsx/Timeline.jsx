import React, { Component } from "react";
import Post from "./Post.jsx";

class Timeline extends Component {

    constructor() {
      super();
    }

    generatePosts() {
      return this.props.timeline.map((post, index) => {
        let options = { month: 'short', day: 'numeric' };
        let createdAt  = new Date(post.createdAt).toLocaleDateString("en-US", options);
        return <Post key={index}
                     profileImageUrl={post.user.profileImageUrl}
                     name={post.user.name}
                     twitterHandle={post.user.twitterHandle}
                     createdAt={createdAt}
                     message={post.message}
                     url={post.url} />;
      });
    }

    render() {
      return (
        <div id="timeline">
          {this.generatePosts()}
        </div>
      );
    }

}

export default Timeline;
