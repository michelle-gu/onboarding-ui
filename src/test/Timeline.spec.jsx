import React from 'react';
import { shallow } from 'enzyme';
import Timeline from '../jsx/Timeline.jsx';
import Post from '../jsx/Post.jsx';

const exampleTimeline = [
    {
        "message": "Hello",
        "user": {
            "twitterHandle": "myHandle",
            "name": "myName",
            "profileImageUrl": "http://pbs.twimg.com/profile_images/0/my_photo.jpg"
        },
        "createdAt": 1568236459000,
        "url": "https://twitter.com/myHandle/status/0"
    },
    {
        "message": "Hi",
        "user": {
            "twitterHandle": "myHandle",
            "name": "myName",
            "profileImageUrl": "http://pbs.twimg.com/profile_images/0/my_photo.jpg"
        },
        "createdAt": 1568236459001,
        "url": "https://twitter.com/myHandle/status/1"
    }
];

describe('Timeline without posts', () => {
    let wrapper;
    beforeEach(() => wrapper = shallow(<Timeline timeline={[]} />));
    
    it('should render 1 <div />', () => {
        expect(wrapper.find('div').length).toEqual(1);
    });

});


describe('Timeline with posts', () => {
    let wrapper;
    beforeEach(() => wrapper = shallow(<Timeline timeline={exampleTimeline} />));

    it('should render 1 <div />', () => {
        expect(wrapper.find('div').length).toEqual(1);
    });

    it('should render number of posts from timeline', () => {
        expect(wrapper.find('Post').length).toEqual(wrapper.instance().props.timeline.length);
    });

    it('posts should be inside div', () => {
        
    });

});
