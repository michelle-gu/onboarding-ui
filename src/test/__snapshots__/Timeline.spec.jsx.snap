// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline with posts should render correctly 1`] = `
<div
  id="timeline"
>
  <Post
    createdAt={1568236459000}
    key="0"
    message="Hello"
    name="myName"
    profileImageUrl="http://pbs.twimg.com/profile_images/0/my_photo.jpg"
    twitterHandle="myHandle"
    url="https://twitter.com/myHandle/status/0"
  />
  <Post
    createdAt={1568236459001}
    key="1"
    message="Hi"
    name="myName"
    profileImageUrl="http://pbs.twimg.com/profile_images/0/my_photo.jpg"
    twitterHandle="myHandle"
    url="https://twitter.com/myHandle/status/1"
  />
</div>
`;

exports[`Timeline without posts should render correctly 1`] = `
<div
  id="timeline"
/>
`;
